### PR TITLE
[react-graphql] Update to documentation and when to use Preload, Prefetch, KeepFresh

### PR DESCRIPTION
**Pull request by oluwatimio**
## Description

The documentation does not explicitly say when we need to use `MyComponent.Preload` or not.
## Type of change
- [x] react-graphql Patch: Documentation fix (non-breaking change which fixes an issue or adds documentation)
- [ ]  Patch: Bug (non-breaking change which fixes an issue)
- [ ]  Minor: New feature (non-breaking change which adds functionality)
- [ ]  Major: Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [ ] I have added a changelog entry, prefixed by the type of change noted above (Documentation fix and Test update does not need a changelog as we do not publish new version)

### CODE REVIEW

File: packages/react-async/README.md [hunk 1]
```diff
@@ -173,6 +173,31 @@ const MyModalComponent = createAsyncComponent({
...
+```tsx
+const MyComponent = createAsyncComponent({
+  load: () => import(/* webpackChunkName: 'MyComponent' */ './MyComponent'),
+  usePreload: () => usePreload(MyQuery),
+});
+
+<MyComponent.Preload />;
+```
+
+When relying on the library by not including usePreload; in this case, we do not need to have.
```
Comment (TheMallen) [inline packages/react-async/README.md:195]: This sentence is a bit unclear and fragemented,  maybe something like:

 When relying on the default behaviour we do not need to render `MyComponent.Preload`

Comment (oluwatimio) [reply packages/react-async/README.md]: Thank you @TheMallen! .. made the change

File: packages/react-async/README.md [hunk 2]
```diff
@@ -173,6 +173,37 @@ const MyModalComponent = createAsyncComponent({
...
+```tsx
+const MyComponent = createAsyncComponent({
+  load: () => import(/* webpackChunkName: 'MyComponent' */ './MyComponent'),
+  usePreload: () => usePreload(MyQuery),
+});
+
+<MyComponent.Preload />;
+```
+
+When relying on the default behaviour we do not need to render `MyComponent.Preload`
```
Comment (attila-berki) [inline packages/react-async/README.md:195]: I would also mention the `defer` property and the fact that it is ignored / overwritten by the `.Preload` / `.Prefetch` components.

**Comment by oluwatimio** [comment]
Just merged master into this branch. Are we good to merge this @TheMallen and @attila-berki ?

**Pull request merged**